src/Aspire.Dashboard/Components/Controls/GridValue.razor.cs

```diff
@@ -61,6 +61,13 @@ private async Task ToggleMaskStateAsync()
     private async Task CopyTextToClipboardAsync(string? text, string id)
         => await JS.InvokeVoidAsync("copyTextToClipboard", id, text, PreCopyToolTip, PostCopyToolTip);
 
-    private static string TrimLength(string? text)
-        => text?.Length > 8 ? text.Substring(0, 8) : text ?? string.Empty;
+    private string TrimLength(string? text)
+    {
+        if (text is not null && MaxDisplayLength is int maxLength && text.Length > maxLength)
+        {
+            return text[..maxLength];
```

Comment:
One day I'll learn to make better usage of the range syntax...

Reply:
The IDE offered me this transformation. I'm also learning it. If it's too confusing, we can call `Substring` (and adjust the IDE suggestion to convert these to ranges.)